test/redis-invalidation.test.js

```diff
@@ -0,0 +1,299 @@
+    t.plan(4)
+    // Setup Fastify and Mercurius
+    const app = fastify()
+    t.teardown(app.close.bind(app))
+    app.register(mercurius, { schema, resolvers })
+    // Setup Cache
+    let failHit = false
+    app.register(cache, {
+      ttl: 100,
+      storage: {
+        type: 'redis',
+        options: { client: redisClient, invalidation: true }
+      },
+      onHit () {
+        if (failHit) t.fail()
+      },
+      policy: {
+        Query: {
+          get: {
+            references: async ({ arg }) => [`get:${arg.id}`, 'gets']
+          },
+          search: {
+            references: async ({ arg }) => [`search:${arg.id}`, 'searchs']
+          }
+        },
+        Mutation: {
+          set: {
+            invalidate: async () => ['gets']
+          }
+        }
+      }
+    })
+    // Run the request and cache it
+    await request({ app, query: '{ get(id: 11) }' })
+    t.equal(
+      await redisClient.get((await redisClient.smembers('r:get:11'))[0]),
+      '"get 11"'
+    )
+    await request({ app, query: '{ get(id: 12) }' })
+    t.equal(
+      await redisClient.get((await redisClient.smembers('r:get:12'))[0]),
+      '"get 12"'
+    )
+    await request({ app, query: '{ search(id: 11) }' })
+    t.equal(
+      await redisClient.get((await redisClient.smembers('r:search:11'))[0]),
+      '"search 11"'
+    )
+    // Request a mutation, invalidate 'gets'
+    await request({ app, query: 'mutation { set(id: 11) }' })
+    // Check the references of 'searchs', should still be there
+    t.equal(
+      await redisClient.get((await redisClient.smembers('r:search:11'))[0]),
+      '"search 11"'
+    )
+    // 'get:11' should not be present in cache anymore,
+    failHit = true
+    // should trigger onMiss and not onHit
+    await request({ app, query: '{ get(id: 11) }' })
+  })
+
+  test('should not throw on invalidation error', async t => {
+    t.plan(3)
+    // Setup Fastify and Mercurius
+    const app = fastify()
+    t.teardown(app.close.bind(app))
+    app.register(mercurius, { schema, resolvers })
+    // Setup Cache
+    app.register(cache, {
+      ttl: 100,
+      storage: {
+        type: 'redis',
+        options: { client: redisClient, invalidation: true }
+      },
+      onError (type, fieldName, error) {
+        t.equal(type, 'Mutation')
+        t.equal(fieldName, 'set')
+        t.equal(error.message, 'Kaboom')
+      },
+      policy: {
+        Query: {
+          get: {
+            references: async ({ arg }) => [`get:${arg.id}`, 'gets']
+          }
+        },
+        Mutation: {
+          set: {
+            invalidate: async () => {
+              throw new Error('Kaboom')
+            }
+          }
+        }
+      }
+    })
+    // Run the request and cache it
+    await request({ app, query: '{ get(id: 11) }' })
+    try {
+      await request({ app, query: 'mutation { set(id: 11) }' })
+    } catch (e) {
+      t.fail()
+    }
+  })
```

Comment:
try/catch is unnecessary here. if the second request fails, it will reject and the test will fail.

Reply:
Thx, fixed!